### PR TITLE
Allow dumped image to be transferred to a criu page server

### PR DESCRIPTION
**Pull request by huikang**
--page-server to specify the IP address of criu page server
--port to specify the port of the criu page server

Docker-DCO-1.1-Signed-off-by: Hui Kang <hkang.sunysb@gmail.com>

### CODE REVIEW

**Comment by huikang** [comment]
This is a resubmission of  #551 as suggested by boucher.

File: criu_opts.go [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
-	TcpEstablished          bool   // checkpoint/restore established TCP connections
-	ExternalUnixConnections bool   // allow external unix connections
-	ShellJob                bool   // allow to dump and restore shell jobs
+	ImagesDirectory         string              // directory for storing image files
+	WorkDirectory           string              // directory to cd and write logs/pidfiles/stats to
+	LeaveRunning            bool                // leave container in running state after checkpoint
+	TcpEstablished          bool                // checkpoint/restore established TCP connections
+	ExternalUnixConnections bool                // allow external unix connections
+	ShellJob                bool                // allow to dump and restore shell jobs
+	Ps                      *CriuPageServerInfo // allow to dump to criu page server
```
Comment (crosbymichael) [inline criu_opts.go]: Maybe `net.Addr` is a better data structure for this? idk what do you think?

**Comment by huikang** [comment]
CriuPageServerInfo is a structure from criurpc/criurpc.pb.go and the same as in criu rpc.
Ross defined the CriuOpts struct. @boucher, what do you think?

**Comment by crosbymichael** [comment]
Ok, that is fine then

**Comment by boucher** [comment]
I have no real opinion. It's just a placeholder for shuttling data around though, so perhaps it makes sense not to have to convert to and from another type.

**Comment by huikang** [comment]
I agree. Thanks for your comments.

- Hui

**Comment by huikang** [comment]
@avagin ok, I will change to that cli format. Thanks.

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**